lib/view_model/settings/trocador_providers_view_model.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:cake_wallet/store/settings_store.dart';
+import 'package:mobx/mobx.dart';
+
+part 'trocador_providers_view_model.g.dart';
+
+class TrocadorProvidersViewModel = TrocadorProvidersViewModelBase with _$TrocadorProvidersViewModel;
+
+abstract class TrocadorProvidersViewModelBase with Store {
+  TrocadorProvidersViewModelBase(this._settingsStore);
+
+  final SettingsStore _settingsStore;
+
+  @computed
+  Map<String, bool> get providerStates => _settingsStore.trocadorProviderStates;
+
+  @action
+  void toggleProviderState(String providerName) {
+    final currentState = providerStates[providerName] ?? false;
+    _settingsStore.trocadorProviderStates[providerName] = !currentState;
+    _settingsStore.saveTrocadorProviderState(providerName, !currentState);
```

Comment:
**Enhancement:**
this is a duplicate and can be enhanced so we dont have to call 2 functions whenever we use this.
either leave the first line and add a reaction in the settings store that will save the state
or remove the first line since the save function already updates the `trocadorProviderStates`

Reply:
fixed